AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginAuthIntegrationTests/AWSDataStoreCategoryPluginAuthIntegrationTests.swift

```diff
@@ -0,0 +1,230 @@
+
+    struct User {
+        let username: String
+        let password: String
+    }
+
+    let amplifyConfigurationFile = "AWSDataStoreCategoryPluginAuthIntegrationTests-amplifyconfiguration"
+    let credentialsFile = "AWSDataStoreCategoryPluginAuthIntegrationTests-credentials"
+    var user1: User!
+    var user2: User!
+
+    let syncReceived = HubPayload.EventName.DataStore.syncReceived
+    let syncStarted = HubPayload.EventName.DataStore.syncStarted
+
+    override func setUp() {
+        Amplify.Logging.logLevel = .verbose
+
+        do {
+            let credentials = try TestConfigHelper.retrieveCredentials(forResource: credentialsFile)
+
+            guard let user1 = credentials["user1"],
+                let user2 = credentials["user2"],
+                let passwordUser1 = credentials["passwordUser1"],
+                let passwordUser2 = credentials["passwordUser2"] else {
+                    XCTFail("Missing credentials.json data")
+                    return
+            }
+
+            self.user1 = User(username: user1, password: passwordUser1)
+            self.user2 = User(username: user2, password: passwordUser2)
+
+            try Amplify.add(plugin: AWSDataStorePlugin(modelRegistration: SocialNoteModelRegistration()))
+            try Amplify.add(plugin: AWSAPIPlugin())
+            try Amplify.add(plugin: AWSCognitoAuthPlugin())
+            let amplifyConfig = try TestConfigHelper.retrieveAmplifyConfiguration(forResource: amplifyConfigurationFile)
+            try Amplify.configure(amplifyConfig)
+
+        } catch {
+            XCTFail("Error during setup: \(error)")
+        }
+        if isSignedIn() {
+            signOut()
+        }
+    }
+
+    override func tearDown() {
+        signOut()
+        Amplify.reset()
+    }
+
+    /// A user can persist data in the local store without signing in. Once the user signs in,
+    /// the sync engine will start and sync the mutations to the cloud. Once the reconcillation is complete, retrieving
+    /// the same data will contain ownerId
+    ///
+    /// - Given: A DataStore plugin configured with SocialNote model containing with auth rules
+    /// - When:
+    ///    - User is not signed in, then user can successfully save a note to local store
+    ///    - User remains signed out, then user can successfully retrieve the saved note, with empty owner field
+    ///    - User signs in, retrieves note, sync engine is started and reconciles local store with the ownerId
+    ///    - The note now it contains the ownerId
+    func testUnauthenticatedSavesToLocalStoreIsReconciledWithCloudStoreAfterAuthentication() throws {
+        let savedLocalNote = saveNote(content: "owner saved note")
+        let queriedNoteOptional = queryNote(byId: savedLocalNote.id)
+        guard let note = queriedNoteOptional else {
+            XCTFail("Faled to query local note")
+            return
+        }
+        XCTAssertNil(note.owner)
```

Comment:
this seems more or less like a sanity case check where we just want to make sure `note.owner`  hasn't been set?  if that is the case.. maybe we should have something like:
```
guard let note.owner == nil else {
   XCTFail("heres what happend..etc.."
    return
}
```